src/In.ProjectEKA.HipService/Startup.cs

```diff
@@ -138,16 +142,22 @@ public Startup(IConfiguration configuration)
                         OnTokenValidated = context =>
                         {
                             const string claimTypeClientId = "clientId";
-                            if (!context.Principal.HasClaim(claim => claim.Type == claimTypeClientId))
+                            var accessToken = context.SecurityToken as JwtSecurityToken;
+                            if (!CheckRoleInAccessToken(accessToken))
                             {
-                                context.Fail($"Claim {claimTypeClientId} is not present in the token.");
+                                context.Fail("Specified role not found.");
                             }
                             else
                             {
-                                context.Request.Headers["X-ConsentManagerID"] =
-                                    context.Principal.Claims.First(claim => claim.Type == claimTypeClientId).Value;
+                                if (!context.Principal.HasClaim(claim => claim.Type == claimTypeClientId))
+                                {
+                                    context.Fail($"Claim {claimTypeClientId} is not present in the token.");
+                                }
+                                else
+                                {
+                                    context.Request.Headers["X-ConsentManagerID"] = context.Principal.Claims.First(claim => claim.Type == claimTypeClientId).Value;
+                                }    
                             }
```

Comment:
Can you make it bit readable?

ServiceCaller? IsValid(Token) => Check it has clientId and Role, if yes, returns ServiceCaller/Token otherwise Returns null.

And handle the response here accordingly.

How does it sound?

Reply:
Done